<launch>
	<!-- planner -->
	<rosparam file="$(find dynamic_obstacle_planner)/cfg/planner.yaml"  ns="/dyn_obs_planner"/>
    <rosparam file="$(find dynamic_obstacle_planner)/cfg/controller.yaml" ns="/dyn_obs_planner"/>
    
    <!-- Specify TURTLEBOT3_MODEL -->
    <env name="TURTLEBOT3_MODEL" value="waffle_pi"/>

    <!-- Arguments -->
    <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
    <!--never change pos params unless spawn point changes-->
    <arg name="x_pos" default="-3.0"/>
    <arg name="y_pos" default="1.0"/>
    <arg name="z_pos" default="0.0"/>
    <arg name="configuration_basename" default="turtlebot3_lds_2d.lua"/>
    <arg name="open_rviz" default="true"/>
    <arg name="map_file" default="$(find dynamic_obstacle_planner)/maps/my_map.yaml"/>

    <!-- TurtleBot3 -->
    <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
        <arg name="model" value="$(arg model)" />
    </include>

    <!-- rviz -->
    <group if="$(arg open_rviz)"> 
        <node pkg="rviz" type="rviz" name="rviz" required="true"
            args="-d $(find dynamic_obstacle_planner)/rviz/demo.rviz"/>
    </group>

    <!-- AMCL -->
    <include file="$(find dynamic_obstacle_planner)/launch/amcl_custom.launch"/>

    <!-- gazebo -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find turtlebot3_gazebo)/worlds/turtlebot3_house.world"/>
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="true"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
    </include>

    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
    <!-- Map server -->
    <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>
    
    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model turtlebot3 -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -param robot_description" />
	<node pkg="dynamic_obstacle_planner" type="dyn_obs_planner_node" name="planner_node" output="screen" />

</launch>
